### PR TITLE
Adding check for Windows version and alternate flags for Windows Vista/7

### LINKED ISSUE #3656: Electron windows v0.35.2 cannot use shell.moveItemToTrash
Method shell.moveItemToTrash work for electron Mac OSX but in electron-v0.35.2-win32-ia32,  this method cannot move item to recycle bin and always return false
// OS: Windows 7

### PR DESCRIPTION
**Pull request by tylergibson**
Fixes #3656 by adding a Windows version check and using a different set of operations flags for Windows Vista/7.

### CODE REVIEW

File: atom/common/platform_util_win.cc [hunk 1]
```diff
@@ -340,13 +341,25 @@ bool MoveItemToTrash(const base::FilePath& path) {
...
   // Elevation prompt enabled for UAC protected files.  This overrides the
   // SILENT, NO_UI and NOERRORUI flags.
-  if (FAILED(pfo->SetOperationFlags(FOF_NO_UI |
-                                    FOF_ALLOWUNDO |
-                                    FOF_NOERRORUI |
-                                    FOF_SILENT |
-                                    FOFX_SHOWELEVATIONPROMPT |
-                                    FOFX_RECYCLEONDELETE)))
-    return false;
+  if(IsWindows8OrGreater()) {
```
Comment (paulcbetts) [inline atom/common/platform_util_win.cc]:
```c
if (base::win::GetVersion() >= base::win::VERSION_WIN8)
```

File: atom/common/platform_util_win.cc [hunk 2]
```diff
@@ -5,6 +5,7 @@
 #include "atom/common/platform_util.h"
 
 #include <windows.h>
+#include <versionhelpers.h>
```
Comment (paulcbetts) [inline atom/common/platform_util_win.cc]:
```c
#include "base/win/windows_version.h"
```

File: atom/common/platform_util_win.cc [hunk 3]
```diff
@@ -340,13 +341,25 @@ bool MoveItemToTrash(const base::FilePath& path) {
...
-    return false;
+  if(IsWindows8OrGreater()) {
+    // Windows 8 introduces the flag RECYCLEONDELETE and deprecates the
+    // ALLOWUNDO in favor of ADDUNDORECORD.
+    if (FAILED(pfo->SetOperationFlags(FOF_NO_UI |
+                                      FOFX_ADDUNDORECORD |
+                                      FOF_NOERRORUI |
+                                      FOF_SILENT |
+                                      FOFX_SHOWELEVATIONPROMPT |
+                                      FOFX_RECYCLEONDELETE)))
```
Comment (zcbenz) [inline atom/common/platform_util_win.cc]: Line ends in whitespace.

File: atom/common/platform_util_win.cc [hunk 4]
```diff
@@ -340,13 +341,25 @@ bool MoveItemToTrash(const base::FilePath& path) {
...
+                                      FOFX_RECYCLEONDELETE)))                                        
+      return false;
+  } else {
+    // For Windows 7 and Vista, RecycleOnDelete is the default behavior.
+    if (FAILED(pfo->SetOperationFlags(FOF_NO_UI |
+                                      FOF_ALLOWUNDO |
+                                      FOF_NOERRORUI |
+                                      FOF_SILENT |
+                                      FOFX_SHOWELEVATIONPROMPT)))
+      return false;
```
Comment (zcbenz) [inline atom/common/platform_util_win.cc]: Line ends in whitespace.

**Comment by tylergibson** [comment]
Thanks @paulcbetts for the pointer to the chromium version check, much cleaner :)

**Comment by zcbenz** [comment]
Awesome! :+1:

**Pull request merged**